core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
+    exp_model = exp_services.get_exploration_by_id(exp_id)
+    state_model = exp_model.states[state]
+    training_data = state_model.get_training_data()
+    return training_data
+
+
+def fetch_next_job():
+    classifier_job_models = (
+        classifier_services.get_all_classifier_training_jobs())
+    classifier_job_models.sort(key=lambda item: item.created_on)
+    valid_job_models = []
+    failed_job_models = []
+    for classifier_job_model in classifier_job_models:
+        if classifier_job_model.status == feconf.TRAINING_JOB_STATUS_NEW:
+            valid_job_models.append(classifier_job_model)
+        if classifier_job_model.status == (
+                feconf.TRAINING_JOB_STATUS_PENDING):
+            if (datetime.datetime.utcnow() - (
+                    classifier_job_model.last_updated) < TTL):
```

Comment:
Wait, why is this in valid_jobs? If its PENDING and time < TTL, for all you know, its already sent to the VM and is yet to be stored. This should not be fetched again unless time > TTL.